community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelTransactionImplementation.java

```diff
@@ -163,12 +167,13 @@ TransactionType upgradeToSchemaTransaction() throws InvalidTransactionTypeKernel
     private final Collection<Command> extractedCommands = new ArrayCollection<>( 32 );
     private final Locks locksManager;
     private final boolean txTerminationAwareLocks;
+    private final boolean deferringLocks;
```

Comment:
we can completely avoid knowledge about deferring locks in `KernelTransactionImplementation` if we will pass instance of statementLocks that was constructed somewhere else.